pkg/bpf/mapreconciler/ops.go

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package mapreconciler
+
+import (
+	"context"
+	"errors"
+
+	"github.com/cilium/cilium/pkg/statedb"
+	"github.com/cilium/cilium/pkg/statedb/reconciler"
+)
+
+type mapOps[KV KeyValue[K, V], K comparable, V any] struct {
+	m Map[K, V]
+}
+
+func newMapOps[KV KeyValue[K, V], K comparable, V any](m Map[K, V]) reconciler.Operations[KV] {
+	return &mapOps[KV, K, V]{m}
+}
+
+// Delete implements reconciler.Operations.
+func (ops *mapOps[KV, K, V]) Delete(ctx context.Context, txn statedb.ReadTxn, entry KV) error {
+	return ops.m.Delete(entry.Key())
+}
+
+// Prune implements reconciler.Operations.
+func (ops *mapOps[KV, K, V]) Prune(ctx context.Context, txn statedb.ReadTxn, iter statedb.Iterator[KV]) error {
+	desiredKeys := statedb.CollectSet(statedb.Map(iter, (KV).Key))
+	var (
+		key   K
+		value V
+		errs  []error
+	)
+	mapIter := ops.m.Iterate()
+	for mapIter.Next(&key, &value) {
+		if !desiredKeys.Has(key) {
+			if err := ops.m.Delete(key); err != nil {
+				errs = append(errs, err)
+			}
+		}
+	}
+	errs = append(errs, mapIter.Err())
+	return errors.Join(errs...)
+}
+
+// Update implements reconciler.Operations.
+func (ops *mapOps[KV, K, V]) Update(ctx context.Context, txn statedb.ReadTxn, entry KV) (bool, error) {
+	// TODO: Not cheap to check if this changed the value. Returning true will mess up the
```

Comment:
Figure out a solution to this.

Reply:
could do:
```
Update(ctx context.Context, txn statedb.ReadTxn, obj Obj, changed *bool) error
```
If `changed` is `nil` then `Update` doesn't need to check if it's changed.
For BPF map reconciler we'd have `Update` do a `Lookup` and compare if `changed != nil`. Full reconciler would then pass this in.
It's little bit ugly though.